helpers.lua

```diff
@@ -1070,7 +1070,7 @@ function redirect(url)
     if not string.starts(url, "/") and not string.starts(url, "http://") and not string.starts(url, "https://") then
         url = "https://"..url
     end
-    local domain = url:match("^https?://([%w%.]*)/?")
+    local domain = url:match("^https?://([^/]+)/?")
```

Comment:
```suggestion
    local domain = ngx.var.host
```
🤔

Reply:
Nah this can't work.

We could also iterate through the `conf["domains"]` table to try to match a known domain